cowprotocol/cowswap
# fix: check chainId validity on the lowest app level

# Summary

Fixes #2705

It doesn't make sense to check `chainId` validity many times in the whole app.
Because of it, there are a couple of new changes:
1. `ChainIdValidator` blocks everything when `chainId` in the wallet provider is invalid
2. `useWalletInfo` always returns a valid `chainId`

### Important
1. Please, review by commits. [The first commit](https://github.com/cowprotocol/cowswap/pull/2726/commits/10a88c067387854f0165fdc1dd004b96a341aef2) contains main changes, the second one is just refactoring with deleting irrelevant code
2. "Please connect your wallet..." is not closable anymore

  # To Test (Swap page / Account page)

1. When switch the network to an unsupported one, then a new screen should be displayed
2. When switch to a supported network, then should display a normal UI
3. Other cases with switching network

<img width="1728" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/8989e6a9-80e4-48db-8b85-b454e12dfdf0">

<img width="500" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/1fd6d992-eea9-4c8a-92a1-ec269cd5783f">
